controller/contacts.controller.js

```diff
@@ -0,0 +1,179 @@
+import {
+  getAllContacts,
+  getContactById,
+  createContact,
+  updateContact,
+  removeContact,
+} from '../service/contacts.service.js';
+import Joi from 'joi';
+
+const get = async (req, res, next) => {
+  try {
+    const results = await getAllContacts();
+    res.json({
+      status: 'success',
+      code: 200,
+      data: {
+        contacts: results,
+      },
+    });
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const getById = async (req, res, next) => {
+  const { id } = req.params;
+  try {
+    const result = await getContactById(id);
+    if (result) {
+      res.json({
+        status: 'success',
+        code: 200,
+        data: { contact: result },
+      });
+    } else {
+      res.status(404).json({
+        status: 'error',
+        code: 404,
+        message: `Not found contact id: ${id}`,
+        data: 'Not Found',
+      });
+    }
+  } catch (e) {
+    console.error(e);
+    next(e);
+  }
+};
+
+const create = async (req, res, next) => {
+  const reqBodySchema = Joi.object({
```

Comment:
Ten obiekt nie powinien być w funkcji bo teraz przy każdym wywołaniu create() tworzy się nowy obiekt i przy obciążonym serwerze będziesz stresował "garbage collector". To samo przy put.

Reply:
Miałem na początku (hw2) jeden obiekt walidacji zmiennych body reqBodySchema - jeszcze poza funkcjami - globalnie dla wszystkich, tylko wtedy nie potrzebowałem klucza favorite. W momencie, jak dodana została funkcja zmiany klucza favorite true/false, a klucz ten miał być wymagany w req.body, nie mogłem już używać tego wcześniej stworzonego (ogólnego dla wszystkich) reqBodySchem. Czy walidację w tym wypadku lepiej odpuścić po stronie serwera, a zajmować się nią na froncie? Czy tworzyć dwa oddzielne obiekty do walidacji - globalnie?

Reply:
Fixed.